enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+---
+title: single-node-developer-cluster-profile
+authors:
+  - "@rkukura"
+reviewers:
+  - "@deads2k"
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-09-18
+last-updated: 2020-10-05
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Single Node Developer Cluster Profile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add a new single-node-developer [cluster
+profile](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+that defines the set of OpenShift components, and the configuration
+thereof, applicable to single-node OpenShift Container Platform
+clusters intended for application development as well as for learning
+and exploring OpenShift, but never for production deployment of
+applications. This cluster profile will be utilized in producing the
+Code Ready Containers (CRC) product that runs in a single VM on a
+developer’s workstation or laptop.
+
+In a cluster deployed using this profile, a single node serves as both
+the cluster’s control plane and as a worker node. The cluster profile
+will support the OpenShift components needed for application
+development, including the operator hub and console. In order to
+minimize system resource requirements, the cluster profile will
+exclude production-oriented features such as high availability of the
+control plane components, machine upgradability, and advanced
+telemetry and logging.
+
+This cluster profile will result in a new topology of OpenShift
+components to be officially supported for its intended usage in the
+CRC product. Once the profile is introduced, all teams developing
```

Comment:
to me supported means we will run a CI job for every PR that is affecting CRC, who is going to create and maintain this job?

Reply:
Under "Test Plan" below, I discuss adding a new CI job that runs conformance tests against a deployment of the single-node-developer profile. We'll need to decide whether this CI job runs periodically, or is triggered by PRs submitted to certain repos.

In addition to this new CI job, if a new feature or configuration item is added to a component during phase 2, then it should be possible to extend an existing CI job in that component's repo to test the new feature or configuration item, just as for any other change being made to the component. This is also described in that section.

Do you see the need for an additional CI job beyond these?